sourcegraph/cody
# feat(audoedit): inline renderer POC

- Implements an experimental autoedits inline render POC. It shows autoedit suggestions using the inline completion item provider (used for autocomplete) and line decorations. 
- I plan to improve this initial version with multiple smaller PR addressing one use case at a time to make the review easier. The main goal of the POC was to verify the number of cases we can cover with this approach, uncover unknown blockers, and show how it looks in practice.
- [Slack thread with more details and screenshots](https://sourcegraph.slack.com/archives/C07F8LLKE06/p1732793588108939).
- Part of [CODY-4413: Inline-renderer: support cases without suggested new lines](https://linear.app/sourcegraph/issue/CODY-4413/inline-renderer-support-cases-without-suggested-new-lines)
- Acceptance and disposal will be handled in follow up PRs separately. The current version relies on the logic of the default renderer. 
- Known bugs will be added to [the Linear project](https://linear.app/sourcegraph/project/cody-autoedits-dogfooding-to-ga-0a577390780d/issues) and tackled as follow-ups too.

## Test plan

- CI 
- Manual testing using cody-eval repo: 
    - Add `"cody.experimental.autoedits.inline-renderer": true,` to your `settings.json` to enable the experimental renderer.
    - `code-matching-eval/edits_experiments/examples/renderer-testing-examples`. Request suggestions from the server at the places where an autoedit is expected and observe that it's rendered somewhat correctly.
